File: javascript/euro_to_dollars.js
/**
 * Convert an amount from euros to US dollars.
 *
 * @param {number} euros - Amount in euros.
 * @param {number} [rate=1.08] - Conversion rate (1 euro = rate dollars). Default is 1.08.
 * @returns {number} Equivalent amount in US dollars.
 */
function euroToDollars(euros, rate = 1.08) {
  if (typeof euros !== "number" || isNaN(euros)) {
    throw new Error("`euros` must be a valid number.");
  }
  if (typeof rate !== "number" || isNaN(rate)) {
    throw new Error("`rate` must be a valid number.");
  }
  return euros * rate;
}

// If run directly with Node, allow simple CLI usage:
if (require.main === module) {
  const readline = require("readline");

  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
  });

  rl.question("Enter amount in euros: ", (eurosInput) => {
    rl.question(
      "Enter conversion rate (press Enter to use default 1.08): ",
      (rateInput) => {
        try {
          const euros = parseFloat(eurosInput);
          const rate = rateInput.trim() ? parseFloat(rateInput) : 1.08;

          if (isNaN(euros) || isNaN(rate)) {
            throw new Error("Please enter valid numeric values for amount and rate.");
          }

          const dollars = euroToDollars(euros, rate);
          console.log(
            `${euros} EUR = ${dollars.toFixed(2)} USD (rate: ${rate})`
          );
        } catch (err) {
          console.error(err.message);
        } finally {
          rl.close();
        }
      }
    );
  });
}

module.exports = { euroToDollars };

File: python/euro_to_dollars.py
def euro_to_dollars(euros: float, rate: float = 1.08) -> float:
    """
    Convert an amount from euros to US dollars.

    :param euros: Amount in euros.
    :param rate: Conversion rate (1 euro = rate dollars). Default is 1.08.
    :return: Equivalent amount in US dollars.
    """
    return euros * rate


if __name__ == "__main__":
    try:
        euros_input = float(input("Enter amount in euros: "))
        rate_input = input("Enter conversion rate (press Enter to use default 1.08): ").strip()

        if rate_input:
            rate_value = float(rate_input)
        else:
            rate_value = 1.08

        dollars = euro_to_dollars(euros_input, rate_value)
        print(f"{euros_input} EUR = {dollars:.2f} USD (rate: {rate_value})")
    except ValueError:
        print("Please enter valid numeric values for amount and rate.")
